File: backend/src/main/java/com/carikerja/backend/model/JobSeeker.java
package com.carikerja.backend.model;

import jakarta.persistence.*;
import lombok.*;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "job_seekers")
public class JobSeeker {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    private User user;

    private String name;
    private String birthDate;
    private String phoneNumber;
    private String email;
    private String location;
    private Integer age;
    private String gender;
    private String aboutMe;
    private Boolean hasWorkExperience;
    private String resume;
    private String portfolioLink;
    private String githubLink;
    private Boolean willingToWorkRemotely;
}

File: frontend/src/types/JobSeeker.ts
export interface JobSeeker {
    id: number;
    userId: number;
    name: string;
    birthDate: Date;
    phoneNumber: string;
    email: string;
    location: string;
    age: number;
    gender: 'male' | 'female';
    aboutMe: string;
    hasWorkExperience: boolean;
    resume: string;
    portfolioLink: string;
    githubLink: string;
    willingToWorkRemotely: boolean;
  }